code/perceptron/data.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+from common import *
+from multiprocessing import Pool
+from gensim.models import Doc2Vec
+
+doc2vec_model_filepath = "../../word2vecModels/doc2vec_300features_40minwords_10context"
+doc2vec_model = Doc2Vec.load(doc2vec_model_filepath, mmap='r')
+doc2vec_model.delete_temporary_training_data(keep_doctags_vectors=False, keep_inference=True)
+
+
+def infer_vector_worker(review):
+  i, (review_id, sentiment, document) = review
+  features = doc2vec_model.infer_vector(document.split())
```

Comment:
U may wanna tokenize using `KaggleWord2VecUtility.review_to_wordlist`, since that's how the Doc2Vec model is trained, instead of just `split`